### PR TITLE
Port multibyte-char-to-unibyte

### PR DESCRIPTION
**Pull request by shanavas786**

### CODE REVIEW

File: rust_src/src/multibyte.rs [hunk 1]
```diff
@@ -191,6 +191,18 @@ fn raw_byte_from_codepoint(cp: Codepoint) -> c_uchar {
     (cp - 0x3F_FF00) as c_uchar
 }
 
+/// Same as the CHAR_TO_BYTE_SAFE macro.
+#[inline]
+pub fn raw_byte_from_codepoint_safe (cp: Codepoint) -> c_uchar {
```
Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: The space before the argument list here is causing failure due to rustfmt.

File: rust_src/src/multibyte.rs [hunk 2]
```diff
@@ -191,6 +191,18 @@ fn raw_byte_from_codepoint(cp: Codepoint) -> c_uchar {
...
+    if cp < 0x80 {
```
Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: Should be 0x100? (And maybe include the comment from the C version.)

File: rust_src/src/multibyte.rs [hunk 3]
```diff
@@ -191,6 +191,18 @@ fn raw_byte_from_codepoint(cp: Codepoint) -> c_uchar {
...
+        cp as c_uchar
+    } else if cp > MAX_5_BYTE_CHAR {
+        (cp - 0x3F_FF00) as c_uchar
```
Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: this should use `multibyte::raw_byte_from_codepoint`

File: rust_src/src/multibyte.rs (hunk 2 above)
Comment (shanavas786) [inline rust_src/src/multibyte.rs]: why 0x100? ASCII_P compares with 0x80

Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: But there's an "if c < 256" in the C source of `multibyte-char-to-unibyte`.

Comment (shanavas786) [inline rust_src/src/multibyte.rs]: Yeah, updated

**Pull request merged**